src/Crocodealer/Core/GitHub.hs

```diff
@@ -0,0 +1,17 @@
+module Crocodealer.Core.GitHub
+       ( UserName (..)
+       , RepoName (..)
+       , FileName (..)
+       ) where
+
+newtype UserName = UserName
+    { unUserName :: Text
+    } deriving (Show, Eq, IsString)
+
+newtype RepoName = RepoName
+    { unRepoName :: Text
+    } deriving (Show, Eq, IsString)
+
+newtype FileName = FileName
+    { unFileName :: Text
+    } deriving (Show, Eq, IsString)
```

Comment:
Let's not derive `IsString` instances. To avoid creating values of unexpected types from string literals. Some people would disagree with me here and they think that adding `IsString` instance is a good practice, but I think that it can be harmful in some cases.